### DISCUSSION TITLE
Upgrade C++ usage in 7.1 to C++17?

### DISCUSSION THREAD

**Discussion by ericjbohm** [UIUC-PPL/charm#3684] [Development]
Most compiler infrastructure has updated to provide adequate support for C++17.  Any reason why we shouldn't move our usage and requirements up from C++11 to C++17?

**Comment by stwhite91** [comment]
Justin had a PR for enabling this as a non-default option: https://github.com/UIUC-PPL/charm/pull/3597

I think that PR should be re-worked to make C++17 the default

**Comment by evan-charmworks** [comment]
We went C++11 in 2017, so going C++17 in 2023 sounds reasonable. We should document how that raises our minimum requirements, but fortunately I only expect that to involve the versions of compilers and maybe operating systems, and not dropping support for any platform altogether. Needs to be verified.
